### PR TITLE
Add recording functionality

### PR DESCRIPTION
**Pull request by francovenica**
Adding a library and functions to record every test
Updated github actions to store the recordings and recover them when done

### CODE REVIEW

File: .github/workflows/test-the-tests.yml [hunk 1]
```diff
@@ -26,3 +26,9 @@ jobs:
         TESTING_ENV: 'dev'
       with:
         args: yarn run test-ci safe-web/read_only
+    - name: Upload artifact
+      if: ${{ always() }}
```
Comment (DiogoSoaress) [inline .github/workflows/test-the-tests.yml:30]: Are we sure we want to record and store the .mp4 at every run? What do you think of saving it conditionally? https://github.com/actions/upload-artifact#conditional-artifact-upload

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]

File: .github/workflows/test-the-tests.yml (hunk 1 above)
Comment (francovenica) [reply .github/workflows/test-the-tests.yml:30]: do you mean upload the file only if the test fails? Idk if we can know that (if there is a flag or something we can check if the test has fail at some point). Also as far as I know, the files will be over written and we can make something that clean those files after a few days or so.
cc @DaniSomoza

Comment (DaniSomoza) [reply .github/workflows/test-the-tests.yml:30]: yep, from my side I think that makes sense to upload it only on failure executions:

```
    if: failure()
```

**Review by francovenica — COMMENTED** [review]
In the adv_options test, the line 158 is failing because it cannot read the text in the modal the way we are passing it, because the text in the modal looks like this 
image

Did we changed the way that text is formed in the past few days, because the test was working last week

**Comment by francovenica** [comment]
The other test that is failing is the safe app list, because of an ipfs name that is expected to return a working app with a real value, but now the url doesn't work
the URL: https://ipfs.io/ipfs/QmfMq8NgxdDfai1qKv7bGbT8DegYX4mNzmpi9AkY1VJZBS/
image
image

**Review by DiogoSoaress — APPROVED** [review]
🚀

**Pull request merged**